Patch-Source: https://github.com/ryanoasis/nerd-fonts/pull/801 (modified)
--
From ab973fd663194385f6d2558b28e7ef79aaaaff2b Mon Sep 17 00:00:00 2001
From: Fini Jastrow <ulf.fini.jastrow@desy.de>
Date: Mon, 10 Jan 2022 08:35:38 +0100
Subject: [PATCH 1/2] patch-em-all: Allow to specify font specific options

[why]
We want to patch some fonts with different font-patcher options.

[how]
Use the config.cfg file that each source font can have to specify one
arbitrary option to the font-patcher calls.

[note]
This is partially commit 9e2bc9a26 from #723.

Signed-off-by: Fini Jastrow <ulf.fini.jastrow@desy.de>
---

diff --git a/bin/scripts/gotta-patch-em-all-font-patcher!.sh b/bin/scripts/gotta-patch-em-all-font-patcher!.sh
index e1a322b16..f3931563c 100755
--- a/bin/scripts/gotta-patch-em-all-font-patcher!.sh
+++ b/bin/scripts/gotta-patch-em-all-font-patcher!.sh
@@ -73,6 +73,7 @@ function patch_font {
   config_dir=$( cd "$( dirname "$f" )" && pwd)
 
   # source the font config file if exists:
+  # fetches for example config_patch_flags
   if [ -f "$config_dir/config.cfg" ]
   then
     # shellcheck source=/dev/null
@@ -114,10 +115,10 @@ function patch_font {
     exit 1
   }
 
-  fontforge -quiet -script ./font-patcher "$f" -q $powerline $post_process --complete --no-progressbars --outputdir "${patched_font_dir}complete/" 2>/dev/null
-  fontforge -quiet -script ./font-patcher "$f" -q -s ${font_config} $powerline $post_process --complete --no-progressbars --outputdir "${patched_font_dir}complete/" 2>/dev/null
-  fontforge -quiet -script ./font-patcher "$f" -q -w $powerline $post_process --complete --no-progressbars --outputdir "${patched_font_dir}complete/" 2>/dev/null
-  fontforge -quiet -script ./font-patcher "$f" -q -s ${font_config} -w $powerline $post_process --complete --no-progressbars --outputdir "${patched_font_dir}complete/" 2>/dev/null
+  fontforge -quiet -script ./font-patcher "$f" -q $powerline $post_process --complete --no-progressbars --outputdir "${patched_font_dir}complete/" $config_patch_flags 2>/dev/null
+  fontforge -quiet -script ./font-patcher "$f" -q -s ${font_config} $powerline $post_process --complete --no-progressbars --outputdir "${patched_font_dir}complete/" $config_patch_flags 2>/dev/null
+  fontforge -quiet -script ./font-patcher "$f" -q -w $powerline $post_process --complete --no-progressbars --outputdir "${patched_font_dir}complete/" $config_patch_flags 2>/dev/null
+  fontforge -quiet -script ./font-patcher "$f" -q -s ${font_config} -w $powerline $post_process --complete --no-progressbars --outputdir "${patched_font_dir}complete/" $config_patch_flags 2>/dev/null
   # wait for this group of background processes to finish to avoid forking too many processes
   # that can add up quickly with the number of combinations
   #wait

From ad9ce55378ee388e7c8237e7907c4d0030a28de7 Mon Sep 17 00:00:00 2001
From: Fini Jastrow <ulf.fini.jastrow@desy.de>
Date: Thu, 3 Mar 2022 16:03:10 +0100
Subject: [PATCH 2/2] Create symbols only font directly from sfd template

[why]
The symbol only fonts Symbols-1000-em Nerd Font Complete.ttf
and 2048-em in `NerdFontsSymbolsOnly/` are generated from
some 'almost' empty source fonts, that are assumable in turn generated
from the sfd font descriptions in `src/glyphs/`?

The process is not documented and we have issues in the generated font
(for example the glyph for capital `E` is defined (and empty) #581 #765).

[how]
Use the existing font definitions from `src/glyphs/*.sfd` directly as
source font. That needs a change in font-patcher because the empty
fonts have no glyphs that can be used to orient the scaling upon. In
that case scale on the source font definitions EM.

Then we need patch-em-all to also patch *.sfd fonts.

And finally we need patch-em-all to take a font specific command line
switch for font-patcher (compare 9e2bc9a26 of #723) to instruct it to
create a ttf rather than a sfd font file.

In the sfd file we additionally set the Panose type.
And the UnderlinePosition is adjusted to match the current patched font.

[note]
Also fix wrong glob pattern in patch-em-all `*.[o,t]tf`. The comma is
for sure some leftover from a '{}' shell pattern, that is not used
anymore.

Signed-off-by: Fini Jastrow <ulf.fini.jastrow@desy.de>
---
diff --git a/bin/scripts/gotta-patch-em-all-font-patcher!.sh b/bin/scripts/gotta-patch-em-all-font-patcher!.sh
index f3931563c..76f055725 100755
--- a/bin/scripts/gotta-patch-em-all-font-patcher!.sh
+++ b/bin/scripts/gotta-patch-em-all-font-patcher!.sh
@@ -54,7 +54,7 @@ fi
 source_fonts=()
 while IFS= read -d $'\0' -r file ; do
   source_fonts=("${source_fonts[@]}" "$file")
-done < <(find "$source_fonts_dir" -iname "$like_pattern*.[o,t]tf" -type f -print0)
+done < <(find "$source_fonts_dir" -iname "$like_pattern*.[ot]tf" -o -iname "$like_pattern*.sfd" -type f -print0)
 
 # print total number of source fonts found
 echo "$LINE_PREFIX Total source fonts found: ${#source_fonts[*]}"
diff --git a/font-patcher b/font-patcher
index 7d063ab24..6ec673ea8 100755
--- a/font-patcher
+++ b/font-patcher
@@ -664,6 +664,17 @@ class font_patcher:
                 self.font_dim['xmax'] = xmax
 
         # Calculate font height
+        if self.font_dim['height'] == 0:
+            # This can only happen if the input font is empty
+            # Assume we are using our prepared templates
+            self.font_dim = {
+                'xmin'  : 0,
+                'ymin'  : -self.sourceFont.descent,
+                'xmax'  : self.sourceFont.em,
+                'ymax'  : self.sourceFont.ascent,
+                'width' : self.sourceFont.em,
+                'height': 0,
+            }
         self.font_dim['height'] = abs(self.font_dim['ymin']) + self.font_dim['ymax']
 
 
diff --git a/src/glyphs/NerdFontsSymbols 1000 EM Nerd Font Complete Blank.sfd b/src/unpatched-fonts/NerdFontsSymbolsOnly/NerdFontsSymbols 1000 EM Nerd Font Complete Blank.sfd
similarity index 93%
rename from src/glyphs/NerdFontsSymbols 1000 EM Nerd Font Complete Blank.sfd
rename to src/unpatched-fonts/NerdFontsSymbolsOnly/NerdFontsSymbols 1000 EM Nerd Font Complete Blank.sfd
index d49553e80..3e8e0659a 100644
--- a/src/glyphs/NerdFontsSymbols 1000 EM Nerd Font Complete Blank.sfd	
+++ b/src/unpatched-fonts/NerdFontsSymbolsOnly/NerdFontsSymbols 1000 EM Nerd Font Complete Blank.sfd	
@@ -6,7 +6,7 @@ Weight: Regular
 Copyright: Copyright (c) 2016, Ryan McIntyre
 Version: 001.000
 ItalicAngle: 0
-UnderlinePosition: -100
+UnderlinePosition: -150
 UnderlineWidth: 50
 Ascent: 800
 Descent: 200
@@ -20,12 +20,13 @@ OS2Version: 0
 OS2_WeightWidthSlopeOnly: 0
 OS2_UseTypoMetrics: 1
 CreationTime: 1480466430
-ModificationTime: 1480467813
+ModificationTime: 1646316690
 PfmFamily: 17
 TTFWeight: 400
 TTFWidth: 5
 LineGap: 90
 VLineGap: 0
+Panose: 2 0 5 9 0 0 0 0 0 0
 OS2TypoAscent: 0
 OS2TypoAOffset: 1
 OS2TypoDescent: 0
diff --git a/src/glyphs/NerdFontsSymbols 2048 EM Nerd Font Complete Blank.sfd b/src/unpatched-fonts/NerdFontsSymbolsOnly/NerdFontsSymbols 2048 EM Nerd Font Complete Blank.sfd
similarity index 93%
rename from src/glyphs/NerdFontsSymbols 2048 EM Nerd Font Complete Blank.sfd
rename to src/unpatched-fonts/NerdFontsSymbolsOnly/NerdFontsSymbols 2048 EM Nerd Font Complete Blank.sfd
index 4c0601b95..25347bd01 100644
--- a/src/glyphs/NerdFontsSymbols 2048 EM Nerd Font Complete Blank.sfd	
+++ b/src/unpatched-fonts/NerdFontsSymbolsOnly/NerdFontsSymbols 2048 EM Nerd Font Complete Blank.sfd	
@@ -6,7 +6,7 @@ Weight: Regular
 Copyright: Copyright (c) 2016, Ryan McIntyre
 Version: 001.000
 ItalicAngle: 0
-UnderlinePosition: -204
+UnderlinePosition: -306
 UnderlineWidth: 102
 Ascent: 1638
 Descent: 410
@@ -20,12 +20,13 @@ OS2Version: 0
 OS2_WeightWidthSlopeOnly: 0
 OS2_UseTypoMetrics: 1
 CreationTime: 1480466430
-ModificationTime: 1480467841
+ModificationTime: 1646316690
 PfmFamily: 17
 TTFWeight: 400
 TTFWidth: 5
 LineGap: 184
 VLineGap: 0
+Panose: 2 0 5 9 0 0 0 0 0 0
 OS2TypoAscent: 0
 OS2TypoAOffset: 1
 OS2TypoDescent: 0
diff --git a/src/unpatched-fonts/NerdFontsSymbolsOnly/config.cfg b/src/unpatched-fonts/NerdFontsSymbolsOnly/config.cfg
new file mode 100755
index 000000000..fe1bf3137
--- /dev/null
+++ b/src/unpatched-fonts/NerdFontsSymbolsOnly/config.cfg
@@ -0,0 +1 @@
+config_patch_flags="--ext ttf"

